inc/functions_upload.php

```diff
@@ -679,6 +679,81 @@ function upload_attachment($attachment, $update_attachment=false)
 	return $ret;
 }
 
+/**
+ * Process adding attachment(s) when the "Add Attachment" button is pressed.
+ *
+ * @param int $pid The ID of the post.
+ * @param array $forumpermission The permissions for the forum.
+ * @param string $attachwhere Search string "pid='$pid'" or "posthash='".$db->escape_string($mybb->get_input('posthash'))."'"
+ * @param string $action Where called from: "newthread", "newreply", or "editpost"
+ */
+function add_attachments($pid, $forumpermissions, $attachwhere, $action=false)
+{
+	global $db, $mybb, $editdraftpid, $lang;
+
+	$ret = array();
+
+	if($forumpermissions['canpostattachments'])
+	{
+		$attachments = array();
+		$fields = array ('name', 'type', 'tmp_name', 'error', 'size');
+
+		$total = count($_FILES['attachments']['name']);
+
```

Comment:
Isn't an empty check required here?

Reply:
In PHP 7.2.7 everything's alright, it evaluates to NULL giving $total = 0
I do not know what happens in PHP 5 though and how it should be written to be crossplatform.